player/lua/console.lua

```diff
@@ -1001,6 +1001,14 @@ function complete(backwards)
                 completion_start_position = s2
             end
 
+            -- Expand ~ in file completion.
+            if completer.list == file_list and hint:find('^~' .. path_separator) then
+                local home = mp.command_native({'expand-path', '~/'})
+                before_cur = before_cur:gsub('~', home, 1)
+                hint = home .. hint:sub(2)
+                completion_start_position = completion_start_position + len_utf8(home) - 1
```

Comment:
I don't think that `len_utf8(home)` is right. If home contains a symbol that's more then 1 byte long, then `completion_start_position` also needs to move by more then 1.

Reply:
Indeed it was right before. I don't know why I thought `cursor` counts unicode characters.